### ISSUE TITLE
Subtitles don't show up with AllowHighResolutionSFD

### ISSUE THREAD

**Issue by nipkownix** [nipkownix/re4_tweaks#36]
It doesn't seem to happen on videos from Separate Ways (maybe because SW uses a different code path), but can be seen on the main game's intro video.

It also only happens if the video is in HD, which doesn't seem to make a lot of sense. This probably means that none of our code is broken, but this code path is expecting the 512x336 resolution for something related to subtitles, maybe?

How to reproduce: simply change the game's language to Spanish on Steam and start a new game.

Edit: looks like they work in the movie browser, but not in-game.

**Comment by nipkownix** [comment]
Hmm, the game is skipping the `MesSet` call at the start of `cSofdec__draw_maybe` when the 1080p .sfd file is playing. Not sure why.

**Comment by emoose** [comment]
Doesn't Sofdec support embedding subs into the actual sfd file? Maybe the game is making use of that, do you know if your 1080p file has the sub data included?

**Comment by nipkownix** [comment]
> Maybe the game is making use of that, do you know if your 1080p file has the sub data included?

Hmm.. it could be, but I did manage to make some phrases of the subtitle appear by noping a `je` instruction, so that implies the sub data is coming from the game. Weird.

**Comment by nipkownix** [comment]
Yeah, got my hands on some cri sdk from the era, and it seems you can embed a subtitle in it as a ".tag" file. All of the documentation is in Japanese, so I'm a bit lost, but I guess the game must be reading the timing information from there.

**Comment by emoose** [comment]
Hm, how come the movie browser is able to play subs then? Does that use a different player to the cSofdec stuff?

**Comment by nipkownix** [comment]
It does use `cSofdec__draw_maybe` for the movie, but the subtitles are coming from somewhere else. You can even notice the position of the subtitles is different between the movie browser and when you actually start a new game.

**Comment by emoose** [comment]
Looks like `ConfigReadINI` is checking for a `subtitle` config.ini entry, does your INI have that set?

**Comment by nipkownix** [comment]
It does, yeah. I can see the Spanish subtitles without any issues using the original .sfd file, but it doesn't show up even if a try to create a new .sfd with the same resolution.

**Comment by emoose** [comment]
Hmm, what was inside the .tag you used? Kinda seems `mwPlyGetSubtitle` returns a string, which is then converted with `strtol` to an int and passed over to MessageControl::MesSet, maybe .tag is only meant to contain some kind of string ID which is then looked up in the users language setting.

**Comment by nipkownix** [comment]
The .tag file is generated from a .txt which looks like this, according to the SDK:

![image](https://user-images.githubusercontent.com/33870163/149824185-edb79c6f-58bb-4d96-8aa2-31442d273fc4.png)

This was my test txt:

```
29970
0, 10, 1998
10, 20, 1998
100, 200, 1998
1000, 2000, 1998
10000, 20000, 1998
150000, 200000, everything seems to be in order
200000, 250000, alrite gorden foreman, ur suit
```

Welp

**Comment by nipkownix** [comment]
Oh wow, after I mentioned it to albert, he noticed the only difference between opening.sfd and opening_pal.sfd is that the _pal version has the subtitle data inside it!

After copying that data to the new HD file.. the subtitles do show up!

![image](https://user-images.githubusercontent.com/33870163/149825030-16d65255-f266-41e2-bf11-33bf4f39d821.png)

This is what the data looks like:

![image](https://user-images.githubusercontent.com/33870163/149825092-62fe5b5d-af04-4565-b39b-03184f628892.png)

Would be nice if someone could make a tool to at least "transfer" the subtitle data from an old file to a new one, but at least we know that it is possible now.

**Comment by emoose** [comment]
Something interesting, seems the newer Sofdec2 encoder can actually still encode to .sfd files, and game seems to render them fine!

Tested with the medianoche.exe encoder from https://archive.org/download/cri-sdk/sf_video_4.6.33_windows_msvc11_lib_4K-ready.zip, just used `medianoche -in=sample.avi -out=test.sfd` and copied the output to opening_pal.sfd.

Not sure whether the encoding from that might be better/worse than the older sfd encoder though - also I don't see any option for including subtitles there, but maybe the Wii criware tools can mux them in or something.

Sadly doesn't seem the medianocheH264.exe can write H264 into .sfd files, pretty sure I saw H264 mentioned in some Sofdec1 code though... ah well.
